LanguageFeatures/Augmentation-libraries/augmenting_constructors_A11_t01_lib.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+/// ...
+///   - Initializer lists are not re-run, they have already executed and
+///     shouldn't be executed twice. The same goes for initializing formals and
+///     super parameters.
+///
+/// @description Checks that when `augmented()` is called in the body of an
+/// augmenting constructor initializer lists are not re-run.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A11_t01.dart';
+
+augment class C {
+  augment C(): x = "Augmented" {
```

Comment:
Check out 'Add initializers to the initializer list ...' in the augmentation spec.

I think the semantics of initializer lists is cumulative: The introductory constructor (if it can indeed have an initializer list) can introduce an initializer list.

If an initializer list has been introduced then augmenting constructor declarations can _add to_ the existing initializer list (otherwise it can introduce the initializer list, as usual).

But I would expect that it is an error to have one initializer list element that initializes `x` (in the introductory or an augmenting declaration) and then another one that also initializes `x` in a later augmentation. Just like `x = 1, x = 1` is an error in an initializer list which is written in one place.

So we want to make sure that _all_ elements of the initializer list of the constructor _definition_ are evaluated before the first body runs, and they are not re-evaluated later on. Side effects can show that this is indeed the case.

Reply:
Rewritten.

Reply:
Thanks!